kernel/src/expressions/mod.rs

```diff
@@ -457,6 +481,14 @@ impl Expression {
         Predicate::distinct(self, other)
     }
 
+    /// Creates a new unary expression
+    pub fn unary(op: UnaryExpressionOp, expr: impl Into<Expression>) -> Self {
+        Self::Unary(UnaryExpression {
+            op,
+            expr: Box::new(expr.into()),
+        })
```

Comment:
```suggestion
        UnaryExpression::new(op, expr).into()
```
?
or, if we don't (want to) have the needed `impl From<UnaryExpression> for Expression`:
```suggestion
        Self::Unary(UnaryExpression::new(op, expr))
```